Automattic/jetpack
# Add tracking support for Dashboard Quick Switcher

Fixes https://github.com/Automattic/wp-calypso/issues/54308

#### Changes proposed in this Pull Request:

* Based on the event design documentation from FG, I've created three new events called `dashboard_quick_switch_link_clicked` with a prefix depending on the source (e.g. wpcom, atomic and Jetpack) that tracks each time a user switches between Calypso and WP-Admin interfaces using the Dashboard Quick Switcher.
* The events are the following ones: wpcom_dashboard_quick_switch_link_clicked, atomic_dashboard_quick_switch_link_clicked and jetpack_dashboard_quick_switch_link_clicked
* My initial approach was to have one single event for all environments, however after reviewing the FG, I've noticed that we have multiple events based on the source of the event. To keep things consistent I've followed the same rule and created the three events.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


Yes, we are creating a tracking event each time a user clicks on the Dashboard Quick Switcher link that switchers between the Calypso and WP-Admin equivalent.

#### Testing instructions:






Simple sites:
- Apply D64238-code patch on your sandbox
- Go to a simple site and go to "Posts"
- Click on the quick switcher link
- Go back to the switcher and click again on the link
- In Tracks interface you should be able to see two new events called `wpcom_dashboard_quick_switch_link_clicked`

Atomic sites:
- Use Jetpack Beta and switch to this branch on your Atomic site.
- Go to "Posts" and click on the quick switcher link
- Go back to the switcher and click again on the link
- In Tracks interface you should be able to see two new events called `atomic_dashboard_quick_switch_link_clicked`

Both Simple and Atomic:
- In Tracks interface, please expand the events and make sure that the following information is available:
-- plan
-- destination
-- current_page
-- user_id
-- blog id
